docs: Add FAQ entry for troubleshooting delivery (#3192)
docker-mailserver/docker-mailserver: docs/content/faq.md

If you got any problems with SPF and/or forwarding mails, give [SRS](https://github.com/roehling/postsrsd/blob/master/README.rst) a try. You enable SRS by setting `ENABLE_SRS=1`. See the variable description for further information.

### Why are my emails not being delivered?

> There are many reasons why email might be rejected: a misconfigured SSL certificate, a TLD or IP address with a bad reputation, misconfigured DNS records, etc.

Reviewer: ```suggestion
There are many reasons why email might be rejected, common causes are:

- Wrong or untrustworthy SSL certificate.
- A TLD (your domain) or IP address with a bad reputation.
- Misconfigured DNS records.
```

> These aren't problems with DMS, so deliverability issues shouldn't be filed on the issue tracker.

Reviewer: ```suggestion
DMS does not manage those concerns, verify they are not causing your delivery problems before reporting a bug on our issue tracker. Resources that can help you troubleshoot:
```

> [mail-tester](https://www.mail-tester.com/) can test your deliverability.
> 
> [helloinbox](https://www.helloinbox.email/) provides a checklist of things to improve your deliverability.

Reviewer: ```suggestion
- [mail-tester](https://www.mail-tester.com/) can test your deliverability.
- [helloinbox](https://www.helloinbox.email/) provides a checklist of things to improve your deliverability.
```
